### PR TITLE
Added 4 Sum Problem and updated readme

### PR DESCRIPTION
**Pull request by kotwani2883**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added 4 Sum Problem

### Checklist

- [x] I've read the contribution guidelines.
- [x ] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2480

### CODE REVIEW

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -201,8 +201,8 @@ _add list here_
 - [Longest common sub sequence](dp/longest_increasing_subsequence.cpp)
 - [Longest increasing subsequence](dp/longest_increasing_subsequence.cpp)
 - [Matrix Chain Multiplication](dp/Matrix_Chain_Multiplication.cpp)
-- [Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)
-- [[Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)]
+- [Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)
+- [[Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)]
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Revert these changes

File: C-Plus-Plus/other/4Sum.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/* Given an array of integers, find anyone combination of four elements in the array whose sum is equal to a given value X. */
+
+#include<bits/stdc++.h>
+using namespace std;
+
+   /* Function to find combination of 4 elements that sum to given sum */
```
Comment (HarshCasper) [inline C-Plus-Plus/other/4Sum.cpp]: Indent the Comment properly

File: C-Plus-Plus/other/4Sum.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
+/* Given an array of integers, find anyone combination of four elements in the array whose sum is equal to a given value X. */
```
Comment (HarshCasper) [inline C-Plus-Plus/other/4Sum.cpp]: Add a more detailed description

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Rest looks good to merge!

**Comment by kotwani2883** [comment]
Hii , I have made the changes and also commited but it is not showing here. Is there any way to commit to a particular pull request.I know its  very childish but since I am a newbie kindly help me!
Thanks in advance.

File: C-Plus-Plus/README.md [hunk 4]
```diff
@@ -201,8 +201,8 @@ _add list here_
 - [Longest common sub sequence](dp/longest_increasing_subsequence.cpp)
 - [Longest increasing subsequence](dp/longest_increasing_subsequence.cpp)
 - [Matrix Chain Multiplication](dp/Matrix_Chain_Multiplication.cpp)
-- [Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)
-- [[Maximum Subarray Sum (Constant Space)](dp/Max_subarray_sum_constant_space(DP).cpp)]
+<!-- - [Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)
+- [[Maximum Subarray Sum (Constant Space)](<dp/Max_subarray_sum_constant_space(DP).cpp>)] -->
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md:205]: Don't comment them out. Just leave them out as it is. Rebase it from the Master

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by kotwani2883** [comment]
Changes have been done

**Pull request closed without merging**